taca/illumina/HiSeqX_Runs.py

```diff
@@ -63,102 +66,111 @@ def _copy_samplesheet(self):
-        with chdir(self.run_dir):
-            # create Demultiplexing dir, this changes the status to IN_PROGRESS
-            if not os.path.exists("Demultiplexing"):
-                os.makedirs("Demultiplexing")
-        with chdir(self.run_dir):
-            if self.lanes_not_10X:
-               cmd_normal = self.generate_bcl_command(self.lanes_not_10X, bcl2fastq_cmd_counter)
-               misc.call_external_command_detached(cmd_normal, with_log_files = True, prefix="demux_{}".format(bcl2fastq_cmd_counter))
-               logger.info(("BCL to FASTQ conversion and demultiplexing started for "
-                   "normal run {} on {}".format(os.path.basename(self.id), datetime.now())))
-               bcl2fastq_cmd_counter += 1
-            if self.lanes_10X:
-               cmd_10X = self.generate_bcl_command(self.lanes_10X, bcl2fastq_cmd_counter, is_10X = True)
-               misc.call_external_command_detached(cmd_10X, with_log_files = True, prefix="demux_{}".format(bcl2fastq_cmd_counter))
-               logger.info(("BCL to FASTQ conversion and demultiplexing started for "
-                   "10X run {} on {}".format(os.path.basename(self.id), datetime.now())))
-               bcl2fastq_cmd_counter += 1
-        return True
+        for sample_type in sample_type_list:
+            # Looking for lanes with multiple masks under the same sample type
+            lane_table = dict()
+            for lane, lane_contents in self.sample_table.items():
+                for sample in lane_contents:
+                    if sample[sample.keys()[0]]['sample_type'] == sample_type:
+                        if lane_table.get(lane):
+                            if sample[sample.keys()[0]]['index_length'] not in lane_table[lane]:
+                                lane_table[lane].append(sample[sample.keys()[0]]['index_length'])
+                        else:
+                            lane_table.update({lane:[sample[sample.keys()[0]]['index_length']]})
+            # Determine the number of demux needed for the same sample type
+            demux_number_with_the_same_sample_type = len(max([v for k, v in lane_table.items()],key=len))
+            # Prepare sub-samplesheets, masks and commands
+            for i in range(0,demux_number_with_the_same_sample_type):
+                bcl2fastq_cmd_counter += i
+                # Prepare sub-samplesheet
+                # A dictionary with lane and sample IDs to include
+                samples_to_include = dict()
+                # A dictionary with lane and index length for generating masks
+                mask_table = dict()
+                for lane, lane_contents in self.sample_table.items():
+                    try:
+                        index_length = lane_table[lane][i]
+                        if mask_table.get(lane):
+                            mask_table[lane] = index_length
+                        else:
+                            mask_table.update({lane:index_length})
```

Comment:
I don't understand this part, but I'm pretty sure these two lines are equivalent.